content/en/build/building-with-lotus-APIs.md

```diff
@@ -0,0 +1,664 @@
+
+    // lotusClient exposes all Lotus APIs
+    const lotusClient = new LotusClient(localConnector);
+    const version = await lotusClient.common.version();
+    console.log(version);
+    ```
+
+2. Run the following command in your project directory:
+
+    ```shell with-output
+    node basicQuery.mjs
+    ```
+
+    This will output something like:
+
+    ```shell
+    {
+      Version: '1.15.0+calibnet+git.0ac1bbc7a',
+      APIVersion: 66816,
+      BlockDelay: 30
+    }
+    ```
+
+If everything is set up correctly, your code will be able to connect to the Lotus node and return the chain data as shown above. Check the [Lotus Troubleshooting](https://lotus.filecoin.io/lotus/manage/troubleshooting/) if you run into any errors here. 
+
+## Get chain data
+
+Blockchain data is a fundamental part of any blockchain network. Using the Lotus API you can query chain data, including tipsets, blocks, messages, network status, storage deals, storage provider details, and more.
+
+Let's look at some common chain data-related packages in the filecoin.js library's LotusClient:
+
+- `LotusClient.common` - get node info
+- `LotusClient.chain` - interact with the blockchain
+- `LotusClient.wallet` - manage wallets on the Lotus node
+- `LotusClient.client` - interact with the storage and retrieval markets as a client
+- `LotusClient.paych` - interact with and manage payment channels
+- `LotusClient.state` - query, inspect, and interact with chain state
+
+In the previous step, we created a `chainDataQuery.mjs` file to demonstrate the basic steps to connect to the Lotus node and initialize a `LotusClient` to query chain data. Next, let's use some more functions that don't require an authentication token.
+
+1. Get current chain head:
+
+   We can add the following code in `chainDataQuery.mjs` to query current head of the chain. 
+
+     ```javascript with-output
+      //Query the current block head
+      const chainHead = await lotusClient.chain.getHead();
+      console.log(chainHead);
+     ```
+   As a result, it will return the current head of the chain, including current block height, a list of blocks, and a list of CIDs.
+      ```shell
+      {
+        Cids: [
+          {'/': 'bafy2bzacecdkonmhngylnnhrk4azkg2wkgcm6cnm5qn5sk4ww5cszjlvkgkd6'},
+          {'/': 'bafy2bzaceaglcpzhd5gfrzdyt7ce3e5asnbfz3s3stbqyxniziny5snewbpbg'}
+          ],
+        Blocks: [
+          {...},
+          # long list of additional blocks
+          ],
+        Height: 781268
+      }
+      ```
+
+2. Get messages in a block
+
+    ```javascript with-output
+    //Query all the messages in a block
+    const tipSet = await lotusClient.chain.getTipSetByHeight(781267);
+    const messages = await lotusClient.chain.getBlockMessages(tipSet.Cids[0]);
+    console.log(messages);
+    ```
+    Using the above code in`chainDataQuery.mjs` , we can also retrieve messages from a certain block height. And the returned message body will look like the following .
+    ```shell
```

Comment:
````suggestion

    ```
    Using the above code in`chainDataQuery.mjs`, we can also retrieve messages from a certain block height. And the returned message body will look like the following.

    ```shell
````